server/src/main/java/org/opensearch/index/IngestionConsumerFactory.java

```diff
@@ -22,16 +21,16 @@
 public interface IngestionConsumerFactory<T extends IngestionShardConsumer, P extends IngestionShardPointer> {
     /**
      * Initialize the factory with the configuration parameters. This method is called once when the factory is created,
-     * and the parameters are parsed from the {@link org.opensearch.cluster.metadata.IngestionSource} in
+     * and the required parameters are provided from the {@link org.opensearch.cluster.metadata.IngestionSource} in
      * {@link  org.opensearch.cluster.metadata.IndexMetadata}.
-     * @param params the configuration parameters to initialize the factory
+     * @param ingestionSource the ingestion source with configuration parameters to initialize the factory
      */
-    void initialize(Map<String, Object> params);
```

Comment:
Good call -- Since `IngestionSource` holds both the universal pull-based ingestion parameters and the implementation-specific `params`, it makes sense to pass the full config to the factory.